### DISCUSSION TITLE
Error "can't evaluate field SHLVL in type template.Data"

### DISCUSSION THREAD

**Discussion by pfmoore** [JanDeDobbeleer/oh-my-posh#4930] [Troubleshoot]
I'm trying to add the shell level into my prompt. But when I do, I'm getting the above error.

As far as I can see, I need an environment variable `SHLVL` in order for this to work, and my shell (powershell) doesn't set that by default. But I would have expected that if I set `$env:SHLVL=1` for instance, I'd get a 1 displayed, rather than an error. But it doesn't seem to do that, even if I rerun my profile. None of the example themes use `SHLVL`, so I can't be sure if I'm using it correctly. But all I'm doing is `"template": "PS {{ .SHLVL }}"` for testing, so it doesn't seem like it's that complicated.

I tried omp debug prompt and it shows

```
[DEBUG] 15:13:50.536 shell.go:Getenv:325 → 2
[TRACE] 15:13:50.536 shell.go:Getenv(SHLVL) - 0s
```

It's not *that* important, as I can always use `.Env.SHLVL`, it just seems weird that it's not working. Am I doing something dumb here?

**Comment by JanDeDobbeleer** [comment]
To access environment variables, you do need to use `.Env.SHLVL`.

**Reply by pfmoore** [reply]
Understood, but [the documentation](https://ohmyposh.dev/docs/configuration/templates#global-properties) mentions a `SHLVL` global property (and there's code that specifically fetches the `SHLVL` variable). Is that incorrect, or am I misreading it?

**Reply by JanDeDobbeleer** [reply]
You are absolutely correct. Seems like I forgot to map that one. https://github.com/JanDeDobbeleer/oh-my-posh/pull/4931
